pootle/static/js/editor/app.js

```diff
@@ -1264,55 +1264,60 @@ PTL.editor = {
       this.units.visibleRowsBefore + this.units.visibleRowsAfter
     );
 
-    if (!canShow) {
-      return { leadIn, leadOut };
-    }
-
-    if (this.units.begin === 0) {
-      leadIn.push(nt(
-        'Found %(total)s unit.',
-        'Found %(total)s units.',
-        this.units.total, { total: this.units.total }
-      ));
-    } else {
-      // there are some units above;
-      // this can only happen when filter is set to 'all'
-      leadIn.push(nt(
-        'There are %(count)s more unit above.',
-        'There are %(count)s more units above.',
-        this.units.total, { count: this.units.begin }
-      ));
+    if (canShow) {
+      if (this.units.begin === 0) {
+        leadIn.push(nt(
+          'Found %(total)s unit.',
+          'Found %(total)s units.',
+          this.units.total, { total: this.units.total }
+        ));
+      } else {
+        // there are some units above;
+        // this can only happen when filter is set to 'all'
+        leadIn.push(nt(
+          'There are %(count)s more unit above.',
+          'There are %(count)s more units above.',
+          this.units.total, { count: this.units.begin }
+        ));
 
-      leadIn.push(t('Reload the page to see them.'));
+        leadIn.push(t('Reload the page to see them.'));
+      }
     }
 
     if (this.units.end === this.units.total) {
-      leadOut.push(t('Congratulations, you went through all units!'));
+      if (showCongratulationsMessage) {
+        leadOut.push(t('Congratulations, you went through all units!'));
+        leadOut.push($('.back-to-browser').html());
```

Comment:
We already have a reference to the back to browser anchor in `this.backToBrowserEl` which we could reuse as `this.backToBrowserEl.outerHTML`. Nevertheless, this was used prior to the editor rework so it was effectively dead code.

What I would suggest instead is to drop that cached element (the element won't be needed often after all), move the `js-back-to-browser` class name next to the `back-to-browser` class (in the `<li>` element), and then query it without jQuery as `q('.js-back-to-browser').innerHTML`.